### PR TITLE
Port #344 security fix

### PR DESCRIPTION
**Pull request by ionut-arm**
Port the security fix related to the nonce dangling pointer during auth
session creation, and fix the other issues flagged by the compiler.

Signed-off-by: Ionut Mihalcea <ionut.mihalcea@arm.com>

### CODE REVIEW

File: tss-esapi/src/context.rs [hunk 1]
```diff
@@ -56,7 +56,7 @@ pub struct Context {
     ),
     /// TCTI context handle associated with the ESYS context.
     /// As with the ESYS context, an optional Mbox wrapper allows the context to be deallocated.
-    tcti_context: TctiContext,
+    _tcti_context: TctiContext,
```
Comment (wiktor-k) [inline tss-esapi/src/context.rs:59]: :thinking: What's the reason for this change? I looked at #344 but I don't see any renaming... or am I looking at wrong PR?

**Review by wiktor-k — APPROVED** [review]
LGTM :+1:

File: tss-esapi/src/context.rs (hunk 1 above)
Comment (ionut-arm) [reply tss-esapi/src/context.rs:59]: The main reason is "the new compilers complain about this, so I needed to make the change to get the CI to pass" 😄

Comment (ionut-arm) [reply tss-esapi/src/context.rs:59]: (doesn't like it because `_tcti_context` is not directly used anywhere, but we need to keep ownership to it because it's used by the ESAPI context to communicate)

**Review by Superhepper — APPROVED** [review]
